### ISSUE TITLE
ca.rg.net: rcynic-cron: argument must be a string or a number, not 'QuerySet'

### ISSUE THREAD

**Issue by randyqx** [dragonresearch/rpki.net#854]
```
ca.rg.net:/etc/rpki/trust-anchors# rpkic version
buildbot-1.0.1484276702
```

```
Traceback (most recent call last):
  File "/usr/lib/rpki/rpkigui-rcynic", line 40, in <module>
    update_cache()
  File "/usr/lib/python2.7/dist-packages/rpki/gui/gui_rpki_cache/util.py", line 298, in update_cache
    process_cache()
  File "/usr/lib/python2.7/dist-packages/django/utils/decorators.py", line 145, in inner
    return func(*args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/rpki/gui/gui_rpki_cache/util.py", line 191, in process_cache
    cb(auth, rpkiobj)
  File "/usr/lib/python2.7/dist-packages/rpki/gui/gui_rpki_cache/util.py", line 91, in process_certificate
    cert.asns.add(issuer.asns.all())
  File "/usr/lib/python2.7/dist-packages/django/db/models/fields/related.py", line 985, in add
    self._add_items(self.source_field_name, self.target_field_name, *objs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/fields/related.py", line 1091, in _add_items
    '%s__in' % target_field_name: new_ids,
  File "/usr/lib/python2.7/dist-packages/django/db/models/query.py", line 679, in filter
    return self._filter_or_exclude(False, *args, **kwargs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/query.py", line 697, in _filter_or_exclude
    clone.query.add_q(Q(*args, **kwargs))
  File "/usr/lib/python2.7/dist-packages/django/db/models/sql/query.py", line 1310, in add_q
    clause, require_inner = self._add_q(where_part, self.used_aliases)
  File "/usr/lib/python2.7/dist-packages/django/db/models/sql/query.py", line 1338, in _add_q
    allow_joins=allow_joins, split_subq=split_subq,
  File "/usr/lib/python2.7/dist-packages/django/db/models/sql/query.py", line 1200, in build_filter
    lookups, value)
  File "/usr/lib/python2.7/dist-packages/django/db/models/fields/related.py", line 1766, in get_lookup_constraint
    root_constraint.add(lookup_class(targets[0].get_col(alias, sources[0]), value), AND)
  File "/usr/lib/python2.7/dist-packages/django/db/models/lookups.py", line 101, in __init__
    self.rhs = self.get_prep_lookup()
  File "/usr/lib/python2.7/dist-packages/django/db/models/lookups.py", line 139, in get_prep_lookup
    return self.lhs.output_field.get_prep_lookup(self.lookup_name, self.rhs)
  File "/usr/lib/python2.7/dist-packages/django/db/models/fields/__init__.py", line 729, in get_prep_lookup
    return [self.get_prep_value(v) for v in value]
  File "/usr/lib/python2.7/dist-packages/django/db/models/fields/__init__.py", line 985, in get_prep_value
    return int(value)
TypeError: int() argument must be a string or a number, not 'QuerySet'
Program /usr/lib/rpki/rpkigui-rcynic exited with status 1
```

**Comment by sraustein** [comment]
`rpkigui-rcynic`, not `rcynic-cron`.  This is Michael's script that populates some corner of the GUI's database with data from the RP code.   Trace shows it departing for Django-land from `rpki.gui.gui_rpki_cache.util.process_certificate()` but doesn't say much about the object which triggered the exception.  From the nature of the Django error I would guess that something triggered an untested code path, because it looks like a "how could this have ever worked?" exception, but that's just a guess.

Is this repeatable?

**Comment by randyqx** [comment]
sometimes once an hour, sometimes every four hours.  started maybe 36 hours ago.

**Comment by randyqx** [comment]
email continues and is frequent
    Warning!  Stale data from external sources on host ca.rg.net.
    data source    : last import
    rcynic_import  : 2017-01-14 03:20:35.639935

**Comment by sraustein** [comment]
On closer examination of the code in question: it is indeed a seldom (never?) tested branch, and one which Michael probably never saw in the wild: RFC 3779 inheritance in a CA certificate.  If I had to guess, adding ARIN's new TAL caused you to find something you never saw before and hit this.

This looks like it should be a simple fix, but I don't really understand this part of Michael's code very well (OK, at all, but general purpose is obvious) so probably will probably need testing.

**Comment by sraustein** [comment]
Probably fix pushed, package should pop out within the hour.  Let me know if this fixes it, changes it, ....

**Comment by randyqx** [comment]
```
ca.rg.net:/root# rpkic version
buildbot-1.0.1484464502

```
and waiting

**Comment by randyqx** [comment]
no love.  same error

**Comment by randyqx** [comment]
i upgraded both ca0.rpki.net and ca.rg.net to
```
ca0.rpki.net:/root# rpkic version
buildbot-1.0.1484464502
```
and now both have the disease

**Comment by sraustein** [comment]
Reproduced on test VM under pdb.  As expected, problem turned out to be an ARIN CA certificate using inheritance.  Haven't bothered figuring out whether they should have done that or not (grey area but probably legal, out of scope here), current issue is our code crashing.

New iteration of rubber chicken voodoo fix.  This one passed testing on VM where I broke it before.  Will see if packaged version still works when it pops out of the buildbot.
